src/vulkan-renderer/imgui.cpp

```diff
@@ -247,19 +246,23 @@ void ImGUIOverlay::update() {
 
     // TODO() Do not allocate memory at runtime!
     if (!m_imgui_mesh) {
-        m_imgui_mesh = std::make_unique<wrapper::MeshBuffer>(m_device, "imgui_mesh_buffer", sizeof(ImDrawVert),
-                                                             imgui_draw_data->TotalVtxCount, sizeof(ImDrawIdx),
-                                                             imgui_draw_data->TotalIdxCount);
+        std::vector<ImDrawVert> imgui_vertices(imgui_draw_data->TotalVtxCount);
+        std::vector<ImDrawIdx> imgui_indices(imgui_draw_data->TotalIdxCount);
```

Comment:
Would it be possible to avoid all of these copies, this could be a real performance killer. Maybe some kind of span would be more suitable?

Reply:
You are right! This is completely unnecessary. An easy solution would be to offer another constructor which only accepts a `std::size_t` as argument. Since the `VertexType` and `IndexType` are specified in that case by the template arguments, it is clear it must be the actual number of elements in the container, not `sizeof(VertexType) * size`. I will add this constructor.

Reply:
Done.